### PR TITLE
[ticket/13142] Before query to list unapproved and deleted posts

### PR DESCRIPTION
**Pull request by brunoais**
https://tracker.phpbb.com/browse/PHPBB3-13142

### CODE REVIEW

File: phpBB/includes/functions_mcp.php [hunk 1]
```diff
@@ -423,6 +423,28 @@ function phpbb_mcp_sorting($mode, &$sort_days, &$sort_key, &$sort_dir, &$sort_by
...
+			*/
+			$vars = array(
+				'sql',
+				'forum_id',
+				'topic_id',
+				'min_time',
+				'visibility_const',
+				'where_sql',
+			);
+			extract($phpbb_dispatcher->trigger_event('core.mcp_sorting_unapproved_deleted_posts_query_before', compact($vars)));
```
Comment (nickvergessen) [inline phpBB/includes/functions_mcp.php]: Event name should start with the function name?

Comment (brunoais) [inline phpBB/includes/functions_mcp.php]: I guess so. The function is called `phpbb_mcp_sorting()`. I just dropped the obvious `phpbb_` prefix

Comment (nickvergessen) [inline phpBB/includes/functions_mcp.php]: okay right

File: phpBB/includes/functions_mcp.php [hunk 2]
```diff
@@ -423,6 +423,28 @@ function phpbb_mcp_sorting($mode, &$sort_days, &$sort_key, &$sort_dir, &$sort_by
...
+			* This event allows you to control the SQL query to retrieve the list of unapproved and deleted posts
+			*
+			* @event core.mcp_sorting_unapproved_deleted_posts_query_before
+			* @var	string	sql					The current SQL search string
+			* @var	int		forum_id			The forum id of the posts the user is trying to access
+			* @var	int		topic_id			The topic id of the posts the user is trying to access
+			* @var	int		min_time			Integer with the minimum post time that the user is searching for
+			* @var	int		visibility_const	Integer with one of the possible ITEM_* constant values
+			* @var	string	where_sql			Extra information included in the WHERE clause. It must end with "WHERE" or "AND" or "OR"
+			* @since 3.1.3-RC1
```
Comment (marc1706) [inline phpBB/includes/functions_mcp.php]: Please change this to 3.1.4-RC1 now

Comment (brunoais) [inline phpBB/includes/functions_mcp.php]: Done

**Comment by brunoais** [comment]
!unset WIP

**Comment by nickvergessen** [comment]
Merged with https://github.com/phpbb/phpbb/compare/e4b442817377...31ae57483057

**Pull request closed without merging**